harness/determined/common/storage/base.py

```diff
@@ -159,3 +162,40 @@ def _list_directory(root: Union[str, os.PathLike]) -> Dict[str, int]:
                 result[rel_path] = os.path.getsize(abs_path)
 
         return result
+
+
+def from_string(shortcut: str) -> StorageManager:
+    def process_query(q: Dict[str, List[str]]) -> Dict[str, str]:
+        return {k: v[0] for k, v in q.items()}
+
+    p: urllib.parse.ParseResult = urllib.parse.urlparse(shortcut)
+    if p.scheme == "":
+        base_path = p.path
+        return storage.SharedFSStorageManager(base_path=base_path)
```

Comment:
Is this too lax?  What if somebody provides the shortcut string:
```python
checkpoint_storage="localhost:1234/a/b/c?key=value#header"
```

Reply:
Or, what if they have a question mark in a filename?

```python
checkpoint_storage="/path/to/weird?name/checkpoints"
```

Reply:
Also, should we support explicit `file:///path/to/checkpoints`-style urls?

Reply:
The case `"localhost:blah/blah/blah"` fails because `urlparse` interprets "localhost" as the scheme, which doesn't match anything we want.
As for the question mark, well, you can create a (Linux) directory with '?' in the name...
I added `file` as a valid scheme.